compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -81,7 +81,7 @@ object PickledQuotes {
 
   /** Unpickle the tree contained in the TastyExpr */
   def unpickleTerm(pickled: String | List[String], typeHole: TypeHole, termHole: ExprHole)(using Context): Tree = {
-    val unpickled = withMode(Mode.ReadPositions)(unpickle(pickled, isType = false))
+    val unpickled = withMode(Mode.ReadPositions)(unpickle(pickled, isType = false)).asInstanceOf[Inlined]
```

Comment:
With this `asInstanceOf` and `@uncheked` bellow I think it is better to refactor the code into a proper pattern match
```scala
withMode(Mode.ReadPositions)(unpickle(pickled, isType = false)) match
    case unpickled @ Inlined(call, Nil, expansion) =>
      val inlineCtx = inlineContext(call)
      val expansion1 = spliceTypes(expansion, typeHole)(using inlineCtx)
      val expansion2 = spliceTerms(expansion1, typeHole, termHole)(using inlineCtx)
      cpy.Inlined(unpickled)(call, Nil, expansion2)